neuraxle/distributed/streaming.py

```diff
@@ -0,0 +1,511 @@
+        name, n_workers, additional_worker_arguments, max_size, actual_step = self._get_step_params(step)
+
+        return QueueWorker(
+            actual_step,
+            n_workers=n_workers,
+            use_threading=self.use_threading,
+            max_size=max_size,
+            additional_worker_arguments=additional_worker_arguments
+        ).set_name('QueueWorker{}'.format(name))
+
+    def _get_step_params(self, step):
+        """
+        Return all params necessary to create the QueuedPipeline for the given step.
+
+        :param step: tuple
+        :type step: QueuedPipelineStepsTupleList
+
+        :return: return name, n_workers, max_size, actual_step
+        :rtype: tuple(str, int, int, BaseStep)
+        """
+        if len(step) == 2:
+            name, actual_step = step
+            max_size = self.max_size
+            n_workers = self.n_workers_per_step
+            additional_arguments = []
+        elif len(step) == 3:
+            name, n_workers, actual_step = step
+            max_size = self.max_size
+            additional_arguments = []
+        elif len(step) == 4:
+            if isinstance(step[2], Iterable):
+                name, n_workers, additional_arguments, actual_step = step
+                max_size = self.max_size
+            else:
+                name, n_workers, max_size, actual_step = step
+                additional_arguments = []
+        elif len(step) == 5:
+            name, n_workers, additional_arguments, max_size, actual_step = step
+        else:
+            raise Exception(
+                'Invalid Queued Pipeline Steps Shape. Please use one of the following: (step_name, n_workers, max_size, step), (step_name, n_workers, step), (step_name, step)')
+
+        return name, n_workers, additional_arguments, max_size, actual_step
+
+    def _will_transform_data_container(self, data_container: DataContainer, context: ExecutionContext) -> (
+            DataContainer, ExecutionContext):
+        """
+        Start the :class:`QueueWorker` for each step before transforming the data container.
+
+        :param data_container: data container
+        :type data_container: DataContainer
+        :param context: execution context
+        :type context: ExecutionContext
+        :return:
+        """
+        self.connect_queue_workers()
+
+        return data_container, context
+
+    def _transform_data_container(self, data_container: DataContainer, context: ExecutionContext) -> DataContainer:
+        """
+        Transform data container
+
+        :param data_container: data container to transform.
+        :type data_container: DataContainer
+        :param context: execution context
+        :type context: ExecutionContext
+        :return: data container
+        """
+        data_container_batches = data_container.convolved_1d(stride=self.batch_size, kernel_size=self.batch_size)
+        n_batches = self.get_n_batches(data_container)
```

Comment:
add test to ensure that 15 items with batch size 10 will take the last 5 as well (last 5 items are the last batch).